lib/reek/smells/unused_private_method.rb

```diff
@@ -0,0 +1,128 @@
+require_relative 'smell_detector'
+require_relative 'smell_warning'
+require_relative '../context/method_context'
+require_relative '../context/singleton_method_context'
+
+module Reek
+  module Smells
+    # Class for storing `hits` which are unused private methods
+    # we found in the given context. `name`, `line` and `scope` are then used to
+    # construct SmellWarnings.
+    class Hit
+      attr_reader :name, :scope, :line
+
+      def initialize(context)
+        @name  = context.name
+        @scope = context.class == Context::SingletonMethodContext ? 'class' : 'instance'
```

Comment:
How about making it a bit more elastic and check for potential `Context::SingletonMethodContext` descendants as well?

```ruby
@scope = context.class < Context::SingletonMethodContext ? 'class' : 'instance'
```